File: app/src/main/java/com/example/mentalhealth/positivenotes/StoryAdapter.java
package com.example.mentalhealth.positivenotes;

import android.content.Context;
import android.view.LayoutInflater;
import android.view.View;
import android.view.ViewGroup;
import android.widget.TextView;

import androidx.annotation.NonNull;
import androidx.recyclerview.widget.RecyclerView;

import java.util.ArrayList;

import mentalhealth.R;

public class StoryAdapter extends RecyclerView.Adapter<StoryAdapter.MYViewHolder> {
   Context context;
   ArrayList<SS> list;

    public StoryAdapter(Context context, ArrayList<SS> list) {
        this.context = context;
        this.list = list;
    }

    @NonNull
    @Override
    public MYViewHolder onCreateViewHolder(@NonNull ViewGroup parent, int viewType) {
        View v = LayoutInflater.from(context).inflate(R.layout.storyentry,parent,false);

        return new MYViewHolder(v);
    }

    @Override
    public void onBindViewHolder(@NonNull MYViewHolder holder, int position) {
                  SS story = list.get(position);
//                  holder.name.setText(story.getName2());
//        holder.email.setText(story.getEmail2());
        holder.story.setText(story.getStory());


    }

    @Override
    public int getItemCount() {
        return list.size();
    }

    public static class MYViewHolder extends  RecyclerView.ViewHolder {
         TextView name, email, story;
        public MYViewHolder(@NonNull View itemView) {
            super(itemView);
//            name = itemView.findViewById(R.id.names);
//            email=itemView.findViewById(R.id.emails);
            story = itemView.findViewById(R.id.storys);


        }
    }
}

File: app/src/main/java/com/example/mentalhealth/adapter/StoryAdapter.kt
package com.example.mentalhealth.adapter

import android.content.Context
import android.view.LayoutInflater
import android.view.View
import android.view.ViewGroup
import android.widget.TextView
import androidx.recyclerview.widget.RecyclerView
import com.example.mentalhealth.model.Story
import mentalhealth.R

class StoryAdapter(val context: Context, val storyList: ArrayList<Story>):
    RecyclerView.Adapter<StoryAdapter.UserViewHolder>() {

    override fun onCreateViewHolder(parent: ViewGroup, viewType: Int): UserViewHolder {
         val view: View= LayoutInflater.from(context).inflate(R.layout.storyitem, parent,false)
        return UserViewHolder(view)
    }

    override fun onBindViewHolder(holder: UserViewHolder, position: Int) {
           val story =  storyList[position]

        holder.story.text = story.feedback
        holder.names.text = story.name
        holder.email.text = story.email
    }



    override fun getItemCount(): Int {
        return storyList.size

    }
    class UserViewHolder(itemView: View) : RecyclerView.ViewHolder(itemView){
           var email: TextView = itemView.findViewById(R.id.email)
           var  names: TextView = itemView.findViewById(R.id.names)
           val story: TextView = itemView.findViewById(R.id.story)
    }
}